### DISCUSSION TITLE
For loop modeling, can Generalized KIC be used to search for loops of varying size?

### DISCUSSION THREAD

**Discussion by tsamdin** [RosettaCommons/rosetta#111] [General]
If I don't know what size (number of residues) will work best to stabilize in a loop, is there a way to have generalized KIC sample different loop sizes?

**Comment by vmullig** [comment]
No, GenKIC can't add or delete geometry.  You'd have to do it with a series of scripts, each for a different size of loop, or with code-that-generates-code.  You could probably also write something in PyRosetta that would do this.  This was always on my list of stuff to do before I stopped working on Rosetta, but Rosetta has such a clunky way of handling changes in pose length that it was never easy or feasible.  The big problem is what happens downstream: your pose numbering will be different depending on the size of loop inserted, so you'd have to figure out how to set stuff up so that it's not affected by that.

**Reply by vmullig** [reply]
I'm working on a successor to Rosetta called Masala, which should accommodate loop length changes much more gracefully.  Unlike Rosetta's Conformation object, which stores a vector of residues, each of which contains a vector of atoms (imposing very strict indexing rules for all of the molecular geometry), Masala has a more accommodating underlying data structure for its molecular geometry: molecular geometry is a set of atoms and a set of bonds, with annotations that can be applied to groups of atoms and/or bonds.  This makes indels much easier to accommodate.

**Reply by tsamdin** [reply]
Thanks for clarifying! Since I'm very new to Rosetta I think I'll end up writing a series of independent scripts each for a different size loop - hopefully reducing the number of mistakes I make overall. 

Masala sounds very cool, I look forward to trying it out once you release it!

**Comment by roccomoretti** [comment]
Marking this as "closed", as I hope your question has been answered. If that assessment is incorrect, feel free to re-open or open up a new discussion with more details.
